drivers/flash/soc_flash_nrf_mram.c

```diff
@@ -111,8 +132,24 @@ static int nrf_mram_write(const struct device *dev, off_t offset, const void *da
 
 	LOG_DBG("write: %p:%zu", (void *)addr, len);
 
-	memcpy((void *)addr, data, len);
+	mram_no_latency_sync_request();
+	for (uint32_t i = 0; i < (len / MRAM_WORD_SIZE); i++) {
+		while (!nrf_mram_ready(addr + (i * MRAM_WORD_SIZE))) {
```

Comment:
This could result in a non-returning loop on old ironside bundles where the mram isn't forced on by the `mram_no_latency_sync_request()` call. Can we have this fault on timeout or something to make the code reasonably compatible on old variants of the bundle>